File: atak/ATAKMapEngine/lib/src/main/java/com/atakmap/map/layer/feature/geometry/opengl/GLPoint.java

package com.atakmap.map.layer.feature.geometry.opengl;

import com.atakmap.annotations.DeprecatedApi;
import com.atakmap.map.layer.feature.geometry.Point;
import com.atakmap.map.opengl.GLMapView;
import com.atakmap.math.PointD;

/** @deprecated use the batch feature renderering framework */
@Deprecated
@DeprecatedApi(since = "4.1", forRemoval = true, removeAt = "4.4")
public class GLPoint extends GLGeometry {

    private final PointD point;
    private PointD vertex;
    private PointD pixel;

    public GLPoint(Point point) {
        super(point);
        
        this.point = new PointD(point.getX(), point.getY(), point.getZ());
        this.vertex = new PointD(0, 0, 0);
        this.pixel = new PointD(0, 0, 0);
    }
    
    public void getVertex(GLMapView view, int vertexType, PointD vertex) {
        if(this.verticesSrid != view.drawSrid) {
            view.scratch.geo.set(this.point.y, this.point.x);
            view.scene.mapProjection.forward(view.scratch.geo, this.vertex);
            this.verticesSrid = view.drawSrid;
        }
        switch(vertexType) {
            case GLGeometry.VERTICES_PIXEL :
                if(this.pixelCoordsVersion != view.drawVersion) {
                    view.scene.forward.transform(this.vertex, this.pixel);
                    this.pixelCoordsVersion = view.drawVersion;
                }
                vertex.x = pixel.x;
                vertex.y = pixel.y;
                vertex.z = pixel.z;
                break;
            case GLGeometry.VERTICES_PROJECTED :
                vertex.x = this.vertex.x;
                vertex.y = this.vertex.y;
                vertex.z = this.vertex.z;
                break;
            default :
                throw new IllegalArgumentException();
        }
    }
}

File: takengine/mapengine/sdk/src/renderer/feature/GLPoint.cpp

#include "core/GeoPoint.h"
#include "feature/Point.h"
#include "renderer/feature/GLPoint.h"
#include "renderer/map/GLMapView.h"

using namespace atakmap::renderer::feature;
using namespace atakmap::renderer;
using namespace atakmap::feature;
using namespace atakmap::math;
using namespace atakmap::core;

GLPoint::GLPoint(atakmap::feature::Point *pt) : GLGeometry(pt),
    point(pt->x, pt->y, pt->z),
    vertex(0, 0, 0),
    pixel(0, 0, 0) { }

GLPoint::~GLPoint() {

}

void GLPoint::getVertex(const atakmap::renderer::map::GLMapView *view, int vertexType, PointD *vertex) {
    if (verticesSrid != view->drawSrid) {
        GeoPoint scratchGeo(point.y, point.x);
        view->scene.projection->forward(&scratchGeo, &this->vertex);
        verticesSrid = view->drawSrid;
    }
    switch (vertexType) {
    case GLGeometry::VERTICES_PIXEL:
        if (pixelCoordsVersion != view->drawVersion) {
            view->scene.forwardTransform.transform(&this->vertex, &pixel);
            pixelCoordsVersion = view->drawVersion;
        }
        vertex->x = pixel.x;
        vertex->y = pixel.y;
        vertex->z = pixel.z;
        break;
    case GLGeometry::VERTICES_PROJECTED:
        vertex->x = this->vertex.x;
        vertex->y = this->vertex.y;
        vertex->z = this->vertex.z;
        break;
    default:
        throw std::invalid_argument("vertexType");
    }
}


